bats/tests/profile/create-profile-output.bats

```diff
@@ -0,0 +1,575 @@
+}
+
+@test 'complains when an invalid output type is specified' {
+    run rdctl create-profile --from-settings --output=cabbage
+    assert_failure
+    assert_output --partial $"received unrecognized '--output FORMAT' option of cabbage; \"plist\" or \"reg\" must be specified"
+}
+
+@test 'generates plist output from settings' {
+    run rdctl create-profile --output plist --from-settings
+    assert_full_setting_plist_output
+}
+
+@test 'verify plutil is ok with the generated plist output' {
+    if ! is_macos; then
+        skip "Test requires the plist utility and only works on macOS"
+    fi
+    run rdctl create-profile --output plist --from-settings
+    assert_success
+    plutil -s - <<<"$output"
+}
+
+@test "don't need a running app for the rest of this test" {
+    rdctl shutdown
+}
+
+complex_json_data() {
+    echo '{"kubernetes": {"enabled": false}, "containerEngine": { "allowedImages": {"patterns": ["abc", "ghi", "def"] } }, "WSL": { "integrations": { "first": true, "second": false } } }'
+}
+
+assert_registry_output_for_maps_and_lists() {
+    assert_success
+    assert_output - <<'EOF'
+Windows Registry Editor Version 5.00
+[HKEY_CURRENT_USER\SOFTWARE\Policies]
+[HKEY_CURRENT_USER\SOFTWARE\Policies\Rancher Desktop]
+[HKEY_CURRENT_USER\SOFTWARE\Policies\Rancher Desktop\defaults]
+[HKEY_CURRENT_USER\SOFTWARE\Policies\Rancher Desktop\defaults\containerEngine]
+[HKEY_CURRENT_USER\SOFTWARE\Policies\Rancher Desktop\defaults\containerEngine\allowedImages]
+"patterns"=hex(7):61,00,62,00,63,00,00,00,67,00,68,00,69,00,00,00,64,00,65,00,66,00,00,00,00,00
+[HKEY_CURRENT_USER\SOFTWARE\Policies\Rancher Desktop\defaults\kubernetes]
+"enabled"=dword:0
+[HKEY_CURRENT_USER\SOFTWARE\Policies\Rancher Desktop\defaults\WSL]
+[HKEY_CURRENT_USER\SOFTWARE\Policies\Rancher Desktop\defaults\WSL\integrations]
+"first"=dword:1
+"second"=dword:0
+EOF
+}
+
+@test 'encodes multi-string values and maps from a file' {
+    run rdctl create-profile --output reg --hive hkcu --input <(complex_json_data)
+    assert_registry_output_for_maps_and_lists
+}
+
+@test 'encodes multi-string values and maps from a json string' {
+    run rdctl create-profile --output reg --hive hkcu --body "$(complex_json_data)"
+    assert_registry_output_for_maps_and_lists
+}
+
+@test 'complains when no input source is specified' {
+    for type in reg plist; do
+        run rdctl create-profile --output $type
+        assert_failure
+        assert_output --partial "no input format specified: must specify exactly one input format of '--input FILE|-', '--body|-b STRING', or '--from-settings'"
+    done
+}
+
+@test 'complains when multiple input sources are specified' {
+    for type in reg plist; do
+        run rdctl create-profile --output $type --input some-file.txt -b moose
+        assert_failure
+        assert_output --partial "too many input format specified: must specify exactly one input format of '--input FILE|-', '--body|-b STRING', or '--from-settings'"
```

Comment:
The same code is executed 4 times, so could be a function:

```suggestion
too_many() {
    run rdctl create-profile "$@"
    assert_failure
    assert_output --partial "too many input format specified: must specify exactly one input format of '--input FILE|-', '--body|-b STRING', or '--from-settings'"
}
```

Then you call it

```bash
        too_many --output $type --input some-file.txt -b moose
        too_many --output $type --input some-file.txt --from-settings
        too_many --output $type --input some-file.txt -b moose --from-settings
        too_many --output $type -b moose --from-settings
```